src/EnergyPlus/DXCoils.cc

```diff
@@ -208,53 +208,42 @@ void SimDXCoil(EnergyPlusData &state,
-            //    CALL CalcDoe2DXCoil(state, DXCoilNum, CompressorOp, FirstHVACIteration,PartLoadRatio), perform air-side calculations
-            CalcDoe2DXCoil(state, DXCoilNum, CompressorOperation::On, FirstHVACIteration, PartLoadRatio, FanOpMode);
-
-        } else if (SELECT_CASE_var == CoilVRF_Cooling) {
-
-            CalcVRFCoolingCoil(
-                state, DXCoilNum, CompressorOperation::On, FirstHVACIteration, PartLoadRatio, FanOpMode, CompCycRatio, _, AirFlowRatio, MaxCap);
-
-        } else if (SELECT_CASE_var == CoilVRF_Heating) {
-
-            CalcDXHeatingCoil(state, DXCoilNum, PartLoadRatio, FanOpMode, AirFlowRatio, MaxCap);
-
-        } else if (SELECT_CASE_var == CoilVRF_FluidTCtrl_Cooling) {
-
-            CalcVRFCoolingCoil_FluidTCtrl(
-                state, DXCoilNum, CompressorOperation::On, FirstHVACIteration, PartLoadRatio, FanOpMode, CompCycRatio, _, _);
-
-        } else if (SELECT_CASE_var == CoilVRF_FluidTCtrl_Heating) {
-
-            CalcVRFHeatingCoil_FluidTCtrl(state, CompressorOp, DXCoilNum, PartLoadRatio, FanOpMode, _, MaxCap);
-
+    switch (state.dataDXCoils->DXCoil(DXCoilNum).DXCoilType_Num) { // Autodesk:OPTIONAL PartLoadRatio, MaxCap used in this block without PRESENT check
+    case CoilDX_CoolingSingleSpeed: {
+        if (present(CoilCoolingHeatingPLRRatio)) {
+            CalcDoe2DXCoil(state, DXCoilNum, CompressorOp, FirstHVACIteration, PartLoadRatio, FanOpMode, _, AirFlowRatio, CoilCoolingHeatingPLRRatio);
         } else {
-            ShowSevereError(state, "Error detected in DX Coil=" + std::string{CompName});
-            ShowContinueError(state, "Invalid DX Coil Type=" + state.dataDXCoils->DXCoil(DXCoilNum).DXCoilType);
-            ShowFatalError(state, "Preceding condition causes termination.");
-        }
+            CalcDoe2DXCoil(state, DXCoilNum, CompressorOp, FirstHVACIteration, PartLoadRatio, FanOpMode, _, AirFlowRatio);
+        }
+    } break;
+    case CoilDX_HeatingEmpirical: {
+        CalcDXHeatingCoil(state, DXCoilNum, PartLoadRatio, FanOpMode, AirFlowRatio);
+    } break;
+    case CoilDX_HeatPumpWaterHeaterPumped:
+    case CoilDX_HeatPumpWaterHeaterWrapped: {
+        //   call the HPWHDXCoil routine to calculate water side performance set up the DX coil info for air-side calcs
+        CalcHPWHDXCoil(state, DXCoilNum, PartLoadRatio);
+        //    CALL CalcDoe2DXCoil(state, DXCoilNum, CompressorOp, FirstHVACIteration,PartLoadRatio), perform air-side calculations
+        CalcDoe2DXCoil(state, DXCoilNum, CompressorOperation::On, FirstHVACIteration, PartLoadRatio, FanOpMode);
+    } break;
+    case CoilVRF_Cooling: {
+        CalcVRFCoolingCoil(
+            state, DXCoilNum, CompressorOperation::On, FirstHVACIteration, PartLoadRatio, FanOpMode, CompCycRatio, _, AirFlowRatio, MaxCap);
+    } break;
+    case CoilVRF_Heating: {
+        CalcDXHeatingCoil(state, DXCoilNum, PartLoadRatio, FanOpMode, AirFlowRatio, MaxCap);
```

Comment:
Should this be VRF?

Reply:
The VRF heating coil model re-uses the old DX heating coil calc routine. The VRF cooling coil has a few different equations so that coil type has it's own calc routine although we could probably incorporate new routine equations into the old if needed.